modules/context/repo.go

```diff
@@ -497,6 +500,8 @@ func getRefNameFromPath(ctx *Context, path string, isExist func(string) bool) st
 func getRefName(ctx *Context, pathType RepoRefType) string {
 	path := ctx.Params("*")
 	switch pathType {
+	case RepoRefAPI:
+		fallthrough
```

Comment:
Is not this same as `case RepoRefLegacy, RepoRefAPI:`? That means you don't need `RepoRefAPI`.

Reply:
Yes, you do; see [line 623](https://github.com/go-gitea/gitea/pull/2841/files#diff-63a3dd20bb1fd6e9042a6f3353965a58R623)

Reply:
@ethantkoenig Thx, got it. `case RepoRefLegacy, RepoRefAPI:` can be used anyway.